avocado/plugins/runners/vmimage.py

```diff
@@ -0,0 +1,148 @@
+import hashlib
+import os
+import sys
+import traceback
+from multiprocessing import set_start_method
+
+from avocado.core.nrunner.app import BaseRunnerApp
+from avocado.core.nrunner.runner import BaseRunner
+from avocado.core.utils import messages
+from avocado.utils import vmimage
+
+
+class VMImageRunner(BaseRunner):
+
+    name = "vmimage"
+    description = "Runner for dependencies of type vmimage"
+
+    def run(self, runnable):
+        try:
+            yield messages.StartedMessage.get()
+
+            provider = runnable.kwargs.get("provider")
+            version = runnable.kwargs.get("version")
+            arch = runnable.kwargs.get("arch")
+
+            if not all([provider, version, arch]):
+                stderr = "Missing required parameters: provider, version, and arch"
+                yield messages.StderrMessage.get(stderr.encode())
+                yield messages.FinishedMessage.get("error")
+                return
+
+            try:
+                yield messages.StdoutMessage.get(
+                    f"Getting VM image for {provider} {version} {arch}".encode()
+                )
+
+                yield messages.StdoutMessage.get(
+                    f"Downloading VM image for: provider={provider}, version={version}, arch={arch}".encode()
+                )
+                try:
+                    from avocado.core.settings import settings
+
+                    cache_dirs = settings.as_dict().get("datadir.paths.cache_dirs")
+                    if not cache_dirs:
+                        raise ValueError("No cache directories configured in settings.")
+                    cache_dir = cache_dirs[0]
+
+                    provider_normalized = provider.lower()
+                    available_providers = [
+                        p.name.lower() for p in vmimage.IMAGE_PROVIDERS
+                    ]
+                    if provider_normalized not in available_providers:
+                        raise ValueError(
+                            f"Provider '{provider}' not found. Available providers: {', '.join(available_providers)}"
+                        )
+
+                    image = vmimage.Image.from_parameters(
+                        name=provider_normalized,
+                        version=version,
+                        arch=arch,
+                        cache_dir=cache_dir,
+                    )
+
+                    image_path = image.get()
+                    if not image_path or not os.path.exists(image_path):
```

Comment:
IMO all those checks on lines 65-91 are not necessary, because the vmimage utility should be robust enough to provide images without mistakes. We only need to catch possible errors. See https://github.com/avocado-framework/avocado/blob/master/avocado/plugins/vmimage.py#L182